### ISSUE TITLE
Metric tags

### ISSUE THREAD

**Issue by levkk** [markstory/rust-statsd#28]
Hey there,

Thank you for writing this, awesome project!

Curious, should there be support for tagging metrics? I could imagine something like this:

```rust
client.incr("some.metric", vec!["service:api"])
```

or maybe even better:

```rust
let client = Client::new("127.0.0.1:8125", "project", vec!["service:api"]).unwrap()
```

and this would add the tag `service` with value `api` to all metrics.

Thanks!

Edit: I think I was confused; tags seem to be a [Datadog ](https://docs.datadoghq.com/developers/dogstatsd/?tab=hostagent) extension of StatsD, and it's not part of the StatsD spec.

**Comment by markstory** [comment]
> I think I was confused; tags seem to be a [Datadog ](https://docs.datadoghq.com/developers/dogstatsd/?tab=hostagent)extension of StatsD, and it's not part of the StatsD spec.

Yes, tags are a datadog extension. I think other statsd like servers support tags in a similar way though. I don't have much time to do the research, but if tags are implemented consistently by a few other statsd collectors/servers then I don't see a reason to not implement them here.

**Comment by minato128** [comment]
@levkk 
This forked lib has constant tagging feature.
https://github.com/minato128/rust-dogstatsd

**Comment by levkk** [comment]
That's amazing, thank you for showing me that @minato128 .

@markstory  Thank you! I think at this point it's clearly a Datadog extension of the protocol. That being said, Datadog is pretty popular so might be worth an optional setting or something similar.

All the best!

**Comment by timotta** [comment]
Hi @markstory , any chance to merge the @minato128 changes to have the  tags?

**Comment by markstory** [comment]
Yeah there's a good chance that if a pull request was made it would get merged and released.
